bundles/org.openhab.binding.iammeter/src/main/java/org/openhab/binding/iammeter/internal/IammeterHandler.java

```diff
@@ -0,0 +1,203 @@
+            logger.trace("Starting refresh handler");
+            String httpMethod = "GET";
+            String url = "http://admin:admin@" + config.host + ":" + config.port + "/monitorjson";
+            String content = "";
+            InputStream stream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+
+            String response = HttpUtil.executeUrl(httpMethod, url, stream, null, timeout);
+            JsonElement iammeterDataElement = new JsonParser().parse(response);
+            JsonObject iammeterData = iammeterDataElement.getAsJsonObject();
+            String keyWord = "Data";
+            boolean bRemoveChannels = false;
+            String channelProfix = "";
+            if (iammeterData.has("data") || (iammeterData.has("Data") && iammeterData.has("SN"))) {
+                bRemoveChannels = true;
+                if (iammeterData.has("data")) {
+                    keyWord = "data";
+                }
+                for (IammeterWEM3080Channel channelConfig : IammeterWEM3080Channel.values()) {
+                    Channel channel = getThing().getChannel(channelConfig.getId());
+                    if (channel != null){
+                        channelProfix = IammeterBindingConstants.THING_TYPE_POWERMETER + ":"
+                                + channel.getUID().getThingUID().getId();
+                        State state = getDecimal(
+                                iammeterData.get(keyWord).getAsJsonArray().get(channelConfig.getIndex()).toString());
+                        updateState(channel.getUID(), state);
+                    }
+                }
+            } else if (iammeterData.has("Datas") && iammeterData.has("SN")) {
+                keyWord = "Datas";
+                for (IammeterWEM3080TChannel channelConfig : IammeterWEM3080TChannel.values()) {
+                    Channel channel = getThing().getChannel(channelConfig.getId());
+                    if (channel != null){
+                        State state = getDecimal(iammeterData.get(keyWord).getAsJsonArray().get(channelConfig.getRow())
+                                .getAsJsonArray().get(channelConfig.getCol()).toString());
+                        updateState(channel.getUID(), state);
+                    }
+                }
+            }
+            if (bRemoveChannels) {
+                if (!bExtraChannelRemoved) {
+                    thingStructureChanged(channelProfix);
+                }
+            }
+            updateStatus(ThingStatus.ONLINE);
+        // Very rudimentary Exception differentiation
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Communication error with the device: " + e.getMessage());
+        } catch (JsonSyntaxException je) {
+            logger.warn("Invalid JSON when refreshing source {}: {}", getThing().getUID(), je.getMessage());
+        } catch (Exception e) {
```

Comment:
You should specify the concrete exception you want to catch, since catching `Exception` catches also all runtime exceptions.

Reply:
edited.